x-pack/plugins/security_solution/public/common/components/matrix_histogram/index.tsx

```diff
@@ -178,93 +104,46 @@ export const MatrixHistogramComponent: React.FC<MatrixHistogramComponentProps> =
   );
 
   const { toggleStatus, setToggleStatus } = useQueryToggle(id);
-  const [querySkip, setQuerySkip] = useState(skip || !toggleStatus);
-  useEffect(() => {
-    setQuerySkip(skip || !toggleStatus);
-  }, [skip, toggleStatus]);
+
   const toggleQuery = useCallback(
     (status: boolean) => {
       setToggleStatus(status);
-      // toggle on = skipQuery false
-      setQuerySkip(!status);
     },
-    [setQuerySkip, setToggleStatus]
+    [setToggleStatus]
   );
 
-  const isChartEmbeddablesEnabled = useIsExperimentalFeatureEnabled('chartEmbeddablesEnabled');
-
-  const matrixHistogramRequest = {
-    endDate,
-    errorMessage,
-    filterQuery,
-    histogramType,
-    indexNames,
-    onError,
-    startDate,
-    stackByField: selectedStackByOption.value,
-    runtimeMappings,
-    isPtrIncluded,
-    skip: querySkip || isChartEmbeddablesEnabled,
-  };
-  const [loading, { data, inspect, totalCount, refetch }] =
-    useMatrixHistogramCombined(matrixHistogramRequest);
-
   const titleWithStackByField = useMemo(
     () => (title != null && typeof title === 'function' ? title(selectedStackByOption) : title),
     [title, selectedStackByOption]
   );
-  const { responses } = useVisualizationResponse({ visualizationId });
+  const { responses: visualizationResponse } = useVisualizationResponse({ visualizationId });
   const subtitleWithCounts = useMemo(() => {
     if (isInitialLoading) {
       return null;
     }
 
     if (typeof subtitle === 'function') {
-      if (isChartEmbeddablesEnabled) {
-        if (!responses || !visualizationResponseHasData(responses)) {
-          return subtitle(0);
-        }
-        const visualizationCount = responses[0].hits.total;
-        return visualizationCount >= 0 ? subtitle(visualizationCount) : null;
-      } else {
-        return totalCount >= 0 ? subtitle(totalCount) : null;
+      if (!visualizationResponse || !visualizationResponseHasData(visualizationResponse)) {
+        return subtitle(0);
       }
+      const visualizationCount = visualizationResponse[0].hits.total;
+      return visualizationCount >= 0 ? subtitle(visualizationCount) : null;
```

Comment:
Could we extract this logic in a `const visualizationTotalCount: number | null = useMemo(...);` and use the value here and also in line 138, to get the `hideHistogram`?